mmv1/third_party/terraform/resources/resource_bigtable_gc_policy.go

```diff
@@ -268,6 +280,70 @@ func resourceBigtableGCPolicyRead(d *schema.ResourceData, meta interface{}) erro
 	return nil
 }
 
+func gcPolicyToGCRuleString(gc bigtable.GCPolicy, topLevel bool) (map[string]interface{}, error) {
+	result := make(map[string]interface{})
+	switch bigtable.GetPolicyType(gc) {
+	case bigtable.PolicyMaxAge:
+		age := gc.(bigtable.MaxAgeGCPolicy).GetDurationString()
+		if topLevel {
+			rule := make(map[string]interface{})
+			rule["max_age"] = age
+			rules := []interface{}{}
+			rules = append(rules, rule)
+			result["rules"] = rules
+		} else {
+			result["max_age"] = age
+		}
+		break
+	case bigtable.PolicyMaxVersion:
+		// bigtable.MaxVersionsGCPolicy is an int.
+		// Not sure why max_version is a float64.
+		// TODO: Maybe change max_version to an int.
+		version := float64(int(gc.(bigtable.MaxVersionsGCPolicy)))
+		if topLevel {
+			rule := make(map[string]interface{})
+			rule["max_version"] = version
+			rules := []interface{}{}
+			rules = append(rules, rule)
+			result["rules"] = rules
+		} else {
+			result["max_version"] = version
+		}
+		break
+	case bigtable.PolicyUnion:
+		result["mode"] = "union"
+		rules := []interface{}{}
+		for _, c := range gc.(bigtable.UnionGCPolicy).Children {
+			gcRuleString, err := gcPolicyToGCRuleString(c, false)
+			if err != nil {
+				return nil, err
+			}
+			rules = append(rules, gcRuleString)
+		}
+		result["rules"] = rules
+		break
+	case bigtable.PolicyIntersection:
+		result["mode"] = "intersection"
+		rules := []interface{}{}
+		for _, c := range gc.(bigtable.IntersectionGCPolicy).Children {
+			gcRuleString, err := gcPolicyToGCRuleString(c, false)
```

Comment:
Recursion! NIT: You may want to put a comment to let future maintainers know that the function calls itself here.